src/Lecture1.hs

```diff
@@ -123,4 +155,24 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+
+lowerThan :: Int -> [Int] -> Int
+lowerThan n list = do
+    if null list then 0
+    else if n < (head list)
+      then 1 + lowerThan n (tail list)
+    else 0 + lowerThan n (tail list)
+
+greaterThan :: Int -> [Int] -> Int
+greaterThan n list = do
+    if null list then 0
+    else if n > (head list)
+      then 1 + greaterThan n (tail list)
+    else 0 + greaterThan n (tail list)
+
+lowerAndGreater :: Int -> [Int] -> String
+lowerAndGreater n list = do
+  let lower = lowerThan n list
+  let greater = greaterThan n list
+  show n ++ " is greater than " ++ show greater ++ " elements and lower than " ++ show lower ++ " elements"
```

Comment:
That's correct! 💯 

However, with this solution, you will walk through the list twice.
You can write it more performant with only one traversal of the list 🙂
It's desirable to write functions that traverse lists only a single time and consume elements only once. In that case, the composition of such functions will have nice streaming and lazy properties.